docs/Configuration.md

```diff
@@ -3,78 +3,110 @@ id: configuration
+
+:::note
+
+Keep in mind that the resulting configuration object must always be JSON-serializable.
+
+:::
 
-Or through JavaScript:
+The configuration file should simply export an object or a function returning an object:
 
 ```js title="jest.config.js"
-// Sync object
-/** @type {import('@jest/types').Config.InitialOptions} */
-const config = {
+module.exports = {
   verbose: true,
 };
 
-module.exports = config;
-
-// Or async function
+// or async function
 module.exports = async () => {
   return {
     verbose: true,
   };
 };
 ```
 
-Or through TypeScript (if `ts-node` is installed):
+Optionally you can use `defineConfig()` helper for type definitions and autocompletion:
+
+<Tabs groupId="examples">
+<TabItem value="js" label="JavaScript">
+
+```js title="jest.config.js"
+const {defineConfig} = require('jest');
+
+module.exports = defineConfig({
+  verbose: true,
+});
+
+// or async function
+module.exports = defineConfig(async () => {
+  const verbose = await asyncGetVerbose();
+
+  return {verbose};
+});
+```
+
+</TabItem>
+
+<TabItem value="ts" label="TypeScript">
 
 ```ts title="jest.config.ts"
-import type {Config} from '@jest/types';
+import {JestConfig, defineConfig} from 'jest';
 
-// Sync object
-const config: Config.InitialOptions = {
+export default defineConfig({
   verbose: true,
-};
-export default config;
+});
 
-// Or async function
-export default async (): Promise<Config.InitialOptions> => {
-  return {
-    verbose: true,
-  };
-};
+// or async function
+export default defineConfig(async () => {
+  const verbose: JestConfig['verbose'] = await asyncGetVerbose();
+
+  return {verbose};
+});
 ```
 
-Please keep in mind that the resulting configuration must be JSON-serializable.
+</TabItem>
+</Tabs>
+
+:::tip
 
-When using the `--config` option, the JSON file must not contain a "jest" key:
+Jest requires [`ts-node`](https://github.com/TypeStrong/ts-node) to be able to read TypeScript configuration files. Make sure it is installed in your project.
 
-```json
+:::
+
+The configuration can be stored in a JSON file as a plain object:
+
+```json title="jest.config.json"
 {
   "bail": 1,
   "verbose": true
 }
 ```
 
-## Options
+Alternatively Jest's configuration can be defined through the `"jest"` key in the `package.json` of your project:
 
-These options let you control Jest's behavior in your `package.json` file. The Jest philosophy is to work great by default, but sometimes you just need more configuration power.
+```json title="package.json"
+{
+  "name": "my-project",
+  "jest": {
+    "verbose": true
+  }
+}
+```
 
-### Defaults
+## Options
 
 You can retrieve Jest's default options to expand them if needed:
 
 ```js title="jest.config.js"
 const {defaults} = require('jest-config');
```

Comment:
Still thinking. Perhaps it would be better to expose `defineJestConfig()`, `JestConfig` and `jestDefaults` from `'jest'`? Or better from `'jest-config'`? It is nice to install only `jest`, in the other hand these are just devDependencies.